doc/guides/writing-tests.md

```diff
@@ -209,6 +209,36 @@ const assert = require('assert');
 const freelist = require('internal/freelist');
 ```
 
+### Assertions
+
+When writing assertions for object comparison, prefer the strict versions,
+namely:
+
+* `assert.strictEqual()` over `assert.equal()`
+* `assert.deepStrictEqual()` over `assert.deepEqual()`
+
+When using `assert.throws()`, if possible, provide the full error message:
+
+```js
+assert.throws(
+  () => {
+    throw new Error('Wrong value');
+  },
+  /^Error: Wrong value$/ // Instead of something like /Wrong value/
+);
+```
+
+### ES.Next features
+
+At the moment we only use a selected subset of ES.Next features in
+JavaScript code under the `lib` directory for performance considerations,
```

Comment:
Nit: Remove `At the moment` and move `for performance considerations` to the beginning:

> For performance considerations, we only use a selected subset...